rgw: improve handling of illformed Swift's container ACLs.

rzarzynski opened the pull request:
Fixes: http://tracker.ceph.com/issues/18796
Signed-off-by: Radoslaw Zarzynski <rzarzynski@mirantis.com>

rzarzynski commented:
Tempest hasn't found any regression here.

rzarzynski commented:
@Jing-Scott: could you please take a look?

Jing-Scott commented on src/rgw/rgw_acl_swift.cc:
```diff
@@ -64,99 +66,123 @@ static bool uid_is_public(const string& uid)
     if ('-' == url_spec[0]) {
-      is_negative = true;
       url_spec = url_spec.substr(1);
+      boost::algorithm::trim_if(url_spec, boost::is_any_of(" \t\r\n"));
```
the `boost::algorithm::trim_if` satisfied what I wanted before. very good!

cbodley replied:
@rzarzynski if these strings are only passed as headers, i don't think we should be trying to handle control characters like `\t\r\n` here. looking at http://docs.openstack.org/developer/swift/overview_acl.html:
> Spaces may occur between elements as shown in the following example:
> 
> `.r : *, .rlistings, 7ec59e87c6584c348b563254aae4c221:*`

rzarzynski replied:
DONE.

Jing-Scott commented on src/rgw/rgw_acl_swift.cc:
```diff
@@ -64,99 +66,123 @@ static bool uid_is_public(const string& uid)
-        ldout(cct, 10) << "normalized referer to url_spec=" << url_spec
-                       << ", is_negative=" << is_negative << dendl;
-      }
+int RGWAccessControlPolicy_SWIFT::add_grants(RGWRados* const store,
```
Yeah! This logic looks more clear and more easy to understand!

rzarzynski commented on src/rgw/rgw_acl_swift.cc:
```diff
@@ -64,99 +66,123 @@ static bool uid_is_public(const string& uid)
          sub.compare(".referrer") == 0;
```
Absolutely.

Jing-Scott commented:
@cbodley @mattbenjamin
I've no write access. So what do you think?

rzarzynski commented:
@Jing-Scott: we definitely need to obtain results from Teuthology first. Just scheduled [a run](http://pulpito.ceph.com:80/rzarzynski-2017-02-04_13:09:50-rgw-wip-rzarzynski-testing---basic-smithi/) together with #13242.

rzarzynski commented:
The Teuthology run is green except one but unrelated failure.

rzarzynski commented:
@cbodley: I've addressed your comment regarding the whitespace characters by switching from `boost::algorithm::trim_if` to plain `boost::algorithm::trim`. Absolutely no other changes here.
